infra/auth0/js/setCustomIdTokenClaims.js

```diff
@@ -0,0 +1,15 @@
+/**
+ * Handler that will be called during the execution of a PostLogin flow.
+ *
+ * @param {Event} event - Details about the user and the context in which they are logging in.
+ * @param {PostLoginAPI} api - Interface whose methods can be used to change the behavior of the login.
+ */
+exports.onExecutePostLogin = async (event, api) => {
+  const namespace = "https://online.ntnu.no"
+  const { gender, studyYear, phoneNumber, allergies } = event.user.app_metadata
```

Comment:
`phone` is in `user_metadata`.

Rule-of-thumb: user-metadata = user can freely change, app-metadata: system relies on it for something and has to be protected

allergies and gender should also be in user-metadata, study-year would be app-metadata. IMO study-year could just be behind a `${namespace}/membership`, which is wholly managed by the membership-service, which could then contain e.g. `study_year`, `field_of_study`, `expiry_data`, etc.

Reply:
> Rule-of-thumb: user-metadata = user can freely change, app-metadata: system relies on it for something and has to be protected

Is there any use case for this except allowing users to update data in `user_metadata` in SPA's? (which auth0 does not recommend btw https://auth0.com/docs/secure/tokens/access-tokens/management-api-access-tokens/get-management-api-tokens-for-single-page-applications)

IMO it's easier to just store everything in `app_metadata`

Reply:
I think it makes sense to split it, I do not think it causes any overhead.

Reply:
Sure, it does not really matter and you already put phone in `user_metadata` so why not just follow their convention.